src/network/ClientConnectionManager.ts

```diff
@@ -71,6 +69,7 @@ import {AddressProvider} from '../connection/AddressProvider';
 import {ClientFailoverConfig} from '../config';
 import {ClusterService} from '../invocation/ClusterService';
 import {SerializationService} from '../serialization/SerializationService';
+import {ConnectionRegistry, ClientState} from './ConnectionRegistry';
```

Comment:
I'd propose to keep at least `ConnectionRegistry` in `ClientConnectionManager.ts`. That's because we need to make sure that only `ClientConnectionManager` is able to access `ConnectionRegistryImpl` class directly in order mutate the state. The same consideration applies to `ClientState` as only `ClientConnectionManager` should be able to mutate it.

Reply:
Alright